app-modules/service-management/src/Filament/Resources/ServiceRequestTypeResource/Pages/ServiceRequestTypeEmailTemplatePage.php

```diff
@@ -111,28 +133,108 @@ public function save(bool $shouldRedirect = true, bool $shouldSendSavedNotificat
-        } else {
-            $data['service_request_type_id'] = $this->getRecord()->id;
-            $data['type'] = $this->type;
+        /** @var ServiceRequestType $record */
+        $record = $this->getRecord();
+
+        if (! ServiceRequestTypeEmailTemplateTabs::active()) {
+            if ($this->template) {
+                $this->template->update($data);
+            } else {
+                $data['service_request_type_id'] = $record->getKey();
+                $data['type'] = $this->type;
 
-            $this->getRecord()?->templates()->create($data);
+                $record->templates()->create($data);
 
-            unset($this->template);
+                unset($this->template);
+            }
+        } else {
+            foreach (ServiceRequestTypeEmailTemplateRole::cases() as $role) {
+                $templateData = $data[$role->value] ?? null;
+
+                if (
+                    ! $templateData ||
+                    (is_null($templateData['subject']) && is_null($templateData['body']))
+                ) {
+                    continue;
+                }
+
+                $template = ServiceRequestTypeEmailTemplate::firstOrNew([
+                    'service_request_type_id' => $record->getKey(),
+                    'type' => $this->type,
+                    'role' => $role,
+                ]);
+
+                if (! $template->exists && (is_null($templateData['subject']) || is_null($templateData['body']))) {
+                    continue;
+                }
+
+                $template->subject = $templateData['subject'] ?? $template->subject;
+                $template->body = $templateData['body'] ?? $template->body;
+
+                $template->save();
+            }
         }
 
         $this->getSavedNotification()->send();
     }
 
+    /** @return array<int, TiptapEditor> */
+    protected function getEmailTemplateFormSchemaForRole(string $role): array
+    {
+        return [
+            TiptapEditor::make('subject')
+                ->label('Subject')
+                ->placeholder('Enter the email subject here...')
+                ->extraInputAttributes(['style' => 'min-height: 2rem; overflow-y:none;'])
+                ->disableToolbarMenus()
+                ->mergeTags(['created', 'updated', 'status', 'assigned to', 'title', 'type'])
+                ->showMergeTagsInBlocksPanel(false)
+                ->helperText('You may use “merge tags” to substitute information about a service request into your subject line. Insert a “{{“ in the subject line field to see a list of available merge tags'),
+
+            TiptapEditor::make('body')
+                ->label('Body')
+                ->placeholder('Enter the email body here...')
+                ->extraInputAttributes(['style' => 'min-height: 12rem;'])
+                ->mergeTags(['created', 'updated', 'status', 'assigned to', 'title', 'type'])
+                ->columnSpanFull(),
+        ];
+    }
```

Comment:
It doesn't look like `$role` is being used here anymore, so maybe remove that param and rename to `getEmailTemplateFormSchema()`?